src/main/scala/mesosphere/marathon/core/task/tracker/impl/steps/UpdateTaskTrackerStepImpl.scala

```diff
@@ -19,19 +19,7 @@ class UpdateTaskTrackerStepImpl @Inject() (
 
   def processUpdate(
     timestamp: Timestamp, appId: PathId, task: MarathonTask, status: TaskStatus): Future[_] = {
-    val taskId = status.getTaskId
 
-    import org.apache.mesos.Protos.TaskState._
-    status.getState match {
-      case TASK_ERROR | TASK_FAILED | TASK_FINISHED | TASK_KILLED | TASK_LOST =>
-        // Remove from our internal list
-        taskTracker.terminated(appId, taskId.getValue)
-
-      case TASK_RUNNING if !task.hasStartedAt => // was staged and is now running
-        taskTracker.running(appId, status)
-
-      case _ =>
-        taskTracker.statusUpdate(appId, status)
-    }
+    taskTracker.statusUpdate(appId, status)
```

Comment:
The code in the taskTracker mostly duplicated the branches in this method. That was silly. In theory, the TaskTracker should guarantee the atomicity of processing these updates so it made sense to move the logic the the TaskTracker.

Killing an unknown task does not relate to updating the state of a task, thus it is handled outside of the TaskTracker. Furthermore, it would introduce a dependency from the TaskTracker to the MesosSchedulerDriver.